src/nvim/autocmd.c

```diff
@@ -1152,6 +1152,7 @@ void aucmd_prepbuf(aco_save_T *aco, buf_T *buf)
     globaldir = NULL;
 
     block_autocmds();  // We don't want BufEnter/WinEnter autocommands.
+    make_snapshot(SNAP_AUCMD_IDX);
```

Comment:
Is this really needed, now what we banned inserting the aucmdwin? saving/restoring window layout was the anti-pattern we aimed at getting rid of in the first place..

Reply:
Tests fail without it. If I remember correctly, the problem is that even if we do not allow `aucmd_win` to be messed around in `win_split_ins`, calling code still ignores the return value and messes with the window layout?

Reply:
So, shouldn't we fix the calling code then?

Reply:
Looks like the other way around, layout is messed with _before_ `win_split_ins`, in preparation for it.

This only seems to be called in places that want to move a window to a different split.
To do this, `win_split_ins` is called with `new_wp != NULL`, to insert an existing window into a new split.
Layout is messed with to remove the old split (otherwise you'd have 2 splits referring to the same window...).

Calling `win_split_ins` first to check its return value beforehand would potentially cause problems with other windows, I think (e.g: you may get `e_noroom` because the old split is still taking up space).

Instead I just test for `aucmd_win` before the only 2 places that mess with the layout.
`win_split` isn't affected as it creates a new split of a new window from `curwin` (`new_wp == NULL`; `aucmd_win` should be allowed to do that).